pkg/render/intrusion_detection.go

```diff
@@ -97,6 +103,7 @@ func IntrusionDetection(cfg *IntrusionDetectionConfiguration) Component {
 
 // IntrusionDetectionConfiguration contains all the config information needed to render the component.
 type IntrusionDetectionConfiguration struct {
+	IntrusionDetction     *operatorv1.IntrusionDetection
```

Comment:
There is a typo in intrusiondetection.

Moreover, it is not possible for this value to be nil, because the controller would not get to this point. Perhaps change it back from a pointer to the value, because there are nil checks throughout this file that are unnecessary.